### PR TITLE
Allow double clicking to modify file attachments before asked, fix lifecycle problems

### LINKED ISSUE #230925: await inside ChatInputPart `forEach` call
This await is inside a forEach and is therefore difficult to detect and could very likely cause a race condition.'

I'll do a fix for this.

### LINKED ISSUE #230926: Image and file attachments cannot be clicked when composing a message
Repro:

- Open chat
- Drag a file as an attachment
- Double click the file, 🐛 nothing happens but it should open the file

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #230926
Fixes #230925

The thumbnail and hover are still not updated when the image is changed, this is tracked in https://github.com/microsoft/vscode/issues/230930

Demo with Luna Paint installed as the default image editor:

Recording 2024-10-09 at 07 25 34

### CODE REVIEW

**Review by justschen — APPROVED** [review]
very cool thanks!

File: src/vs/workbench/contrib/chat/browser/chatInputPart.ts [hunk 1]
```diff
@@ -656,16 +661,21 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 		onDidChangeCursorPosition();
 	}
 
-	private initAttachedContext(container: HTMLElement, isLayout = false) {
+	private async initAttachedContext(container: HTMLElement, isLayout = false) {
+		this.attachedContextDisposables.clear();
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/browser/chatInputPart.ts:665]: This is redundant right?

File: src/vs/workbench/contrib/chat/browser/chatInputPart.ts [hunk 2]
```diff
@@ -730,16 +744,43 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
...
 			}
 
+			await Promise.all(attachmentInitPromises);
+			if (store.isDisposed) {
+				return;
+			}
+
+			if (file) {
+				widget.style.cursor = 'pointer';
+				store.add(dom.addDisposableListener(widget, dom.EventType.CLICK, (e: MouseEvent) => {
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/browser/chatInputPart.ts:754]: Title says double-click but this is just a single click? Need keyboard handling too and this should maybe become a button or something

File: src/vs/workbench/contrib/chat/browser/chatInputPart.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/chat/browser/chatInputPart.ts:665]: Yes you're right, it's cleared out in the .value set

File: src/vs/workbench/contrib/chat/browser/chatInputPart.ts (hunk 2 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/chat/browser/chatInputPart.ts:754]: I was just copying what was already in the rendered version, created https://github.com/microsoft/vscode/issues/230951 to follow up

**Pull request merged**